drivers/serial/uart_altera_jtag.c

```diff
@@ -58,16 +133,21 @@ static void uart_altera_jtag_poll_out(const struct device *dev,
 	while (!(sys_read32(config->base + UART_ALTERA_JTAG_CTRL_OFFSET) & UART_WFIFO_MASK)) {
 	}
 
-	uint32_t data_val = sys_read32(config->base + UART_ALTERA_JTAG_DATA_OFFSET);
-
-	data_val &= ~UART_DATA_MASK;
-	data_val |= c;
-	sys_write32(data_val, config->base + UART_ALTERA_JTAG_DATA_OFFSET);
+	sys_write8(c, config->base + UART_ALTERA_JTAG_DATA_OFFSET);
```

Comment:
Are you sure this still works on nios2, as you have removed the masking?

Reply:
Hi @dcpleung thank you for the feedback.
I have verified this with niosv.
According to the [user guide](https://www.intel.com/content/www/us/en/docs/programmable/683130/22-3/data-register-48671.html) the data register bit 15 to 32 is read only.
So the masking is not necessary.
I have also change to sys_write8 (instead of 32), so that only 8 bits are written.
I will test with nios2 and get back to you.

Reply:
Hi, 
i have tested with nios2 by using the following samples:

- hello_world
- synchronization
- philosophers

The printk work as expected.

The following overlay is needed, as jtag uart is not enabled by default.
in prj.conf:
`CONFIG_UART_ALTERA_JTAG=y`
in altera_max10.overlay:
```
/ {
     chosen {
         zephyr,console = &jtag_uart;
         zephyr,shell-uart = &jtag_uart;
     };
};

&jtag_uart {
    status = "okay";
};
```